src/WebJobs.Script.WebHost/Security/KeyManagement/ScriptSecrets.cs

```diff
@@ -48,5 +53,16 @@ protected ScriptSecrets()
         public virtual bool RemoveKey(Key item, string keyScope) => GetKeys(keyScope)?.Remove(item) ?? false;
 
         public virtual Key GetFunctionKey(string name, string keyScope) => GetKeys(keyScope)?.FirstOrDefault(k => string.Equals(k.Name, name, StringComparison.OrdinalIgnoreCase));
+
+        public void SaveDecryptonKeyHash()
```

Comment:
This API is not super clear. Can you just update the `DecryptionKey` property to generate this hash if one isn't set?